shared/src/main/scala/LiterateLexer.scala

```diff
@@ -103,113 +127,125 @@ object LiterateLexer extends RegexParsers:
     rep(
-      number | string | singleCharString | comment | rawCode | list | lambdaBlock | normalGroup | unpackVar | varGet | varSet | augVar | word | branch
+      number | string | singleCharString | comment | rawCode | list | lambdaBlock | normalGroup |
+        unpackVar | varGet | varSet | augVar | word | branch | newline | tilde | colon |
+        comma
     )
   )
 
   def apply(code: String): Either[VyxalCompilationError, List[LiterateToken]] =
     (parse(tokens, code): @unchecked) match
       case NoSuccess(msg, next)  => Left(VyxalCompilationError(msg))
       case Success(result, next) => Right(result)
-end LiterateLexer
 
-def recHelp(token: Object): String =
-  token match
-    case Word(value)        => value
-    case AlreadyCode(value) => value
-    case LitComment(value)  => value
-    case Number(value)      => value
-    case Variable(value)    => value
-    case LambdaBlock(value) => value.map(recHelp).mkString("λ", " ", "}")
-    case ListToken(value)   => value.map(recHelp).mkString("[", "|", "]")
-    case value: String      => value
-
-def sbcsify(tokens: List[LiterateToken]): String =
-  val out = StringBuilder()
-
-  for i <- tokens.indices do
-    val token = tokens(i)
-    val next = if i == tokens.length - 1 then None else Some(tokens(i + 1))
+  def recHelp(token: Object): String =
     token match
-      case Word(value) =>
-        out.append(
-          literateModeMappings.getOrElse(
-            value,
-            hardcodedKeywords.getOrElse(value, value)
+      case Word(value)        => value
+      case AlreadyCode(value) => value
+      case LitComment(value)  => value
+      case Number(value)      => value
+      case Variable(value)    => value
+      case LambdaBlock(value) => value.map(recHelp).mkString("λ", " ", "}")
+      case ListToken(value)   => value.map(recHelp).mkString("[", "|", "]")
+      case Newline(value)     => value
+      case value: String      => value
+
+  def sbcsify(tokens: List[LiterateToken]): String =
+    val out = StringBuilder()
+
+    for i <- tokens.indices do
+      val token = tokens(i)
+      val next = if i == tokens.length - 1 then None else Some(tokens(i + 1))
+      token match
+        case Word(value) =>
+          out.append(
+            literateModeMappings.getOrElse(
+              value,
+              hardcodedKeywords.getOrElse(value, value)
+            )
           )
-        )
-      case Number(value) =>
-        if value == "0" then out.append("0")
-        else
+        case Number(value) =>
+          if value == "0" then out.append("0")
+          else
+            next match
+              case Some(Number(nextNumber)) =>
+                if nextNumber == "." && value.endsWith(".") then
+                  out.append(value)
+                else out.append(value + " ")
+              case Some(Group(items)) =>
+                if items.length == 1 &&
+                  LiterateLexer(items.head.toString)
```

Comment:
Perhaps we should instead use `LiterateLexer(value + item.head.toString)` to check if it's interpreted as a single number when the two are together. Otherwise, value might be `1.1` and `item.head` might be `.`, so it might have been totally fine for them to not have a space in between but this would add an unnecessary space. However, if we do use `LiterateLexer(value + item.head.toString)`, we will have to check that the parsed number isn't just `value` but `value` and some part of `item.head` before adding a space.